rfc#00003.md

```diff
@@ -0,0 +1,221 @@
+running on it, all workloads that run on the host can only exist within
+the host trust domain.
+
+### Enarx use case states
+Note: that exact state details may vary by implementation.
+
+#### State 0 Enarx use case (initial state)
+Initial state (we assume no compromised processes on host):
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Enarx host agent
+  - Enarx runtime image
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Enarx client agent
+  - Tenant workload image
+  - Tenant workload image repository
+  
+This is the same as the initial state for the non-Enarx-enabled
+use case, but with the Enarx components added here.
+
+#### State 1 Enarx use case (TODO: check name - pre-attestation)
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Enarx host agent
+  - TEE instance
+  - Enarx runtime instance (within TEE instance)
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Enarx client agent
+  - Tenant workload image
+  - Tenant workload image repository
+  
+The state change from `0` to `1` creates a TEE instance and loads
+the Enarx runtime image into it, running it to create an Enarx
+runtime instance.
+
+#### State 2 Enarx use case (TODO: check name - attested)
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Enarx host agent
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Enarx client agent
+  - Empty Keep
+  - Tenant workload image
+  - Tenant workload image repository
+  
+The state change from `1` to `2` moves the Empty Keep into the tenant
+trust domain.  The TEE instance has transformed into a Keep as
+it has been loaded with the Enarx runtime and attested.
+
+#### State 3 Enarx use case (TODO: check name - provisioned)
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Enarx host agent
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Enarx client agent
+  - Provisioned Keep
+  - Tenant workload image
+  - Tenant workload image repository
+  
+The state change from `2` to `3` provisions the tenant workload into
+the (previously empty) Keep.
+
+
+### Applications and trust domains in Enarx
+It may be that the tenant workload, once running, should not
+continue to exist in the top-level tenant trust domain, but may
+be moved into a sub-domain, or even transferred to another owner or
+tenant.  Such chances are application-level: the key aim of Enarx,
+from a trust domain point of view, is to allow the provisioning of
+workloads by a tenant into their own trust domain, rather than the
+host trust domain.
+
+## Reference
+
+rfc#00002
+
+## Drawbacks
+
+n/a
+
+## Rationale and alternatives
+
+n/a
+
+## Prior art
+
+n/a
+
+## Unresolved questions
+
+n/a
+
+## Implementations
+
+None.
+
+## Future Possibilities
+
+- Threat models should be documented and published as RFCs (Issue 15)[https://github.com/enarx/rfcx/issues/15]
```

Comment:
Switch parens and square brackets for this markdown link.

Also, instead of using "Issue 15" for a link text, just do:

```markdown
- [Thread models](https://github.com/enarx/rfcx/issues/15) should be documented and published through the RFC process.